package net.rubenmartinez.rhe.app.controller.exception;

/**
 * A {@code RuntimeException} to signal an error when a user tries to 
 * do a not allowed operation, such as accessing a TodoDTO not belonging to him
 */
public class UserNotAllowedException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    /** Constructs a new runtime exception with the specified detail message.
     * The cause is not initialized, and may subsequently be initialized by a
     * call to {@link #initCause}.
     *
     * @param   message   the detail message. The detail message is saved for
     *          later retrieval by the {@link #getMessage()} method.
     */
    public UserNotAllowedException(String message) {
        super(message);
    }

    /**
     * Constructs a new runtime exception with the specified detail message and
     * cause.  <p>Note that the detail message associated with
     * {@code cause} is <i>not</i> automatically incorporated in
     * this runtime exception's detail message.
     *
     * @param  message the detail message (which is saved for later retrieval
     *         by the {@link #getMessage()} method).
     * @param  cause the cause (which is saved for later retrieval by the
     *         {@link #getCause()} method).  (A <tt>null</tt> value is
     *         permitted, and indicates that the cause is nonexistent or
     *         unknown.)
     * @since  1.4
     */
    public UserNotAllowedException(String message, Throwable cause) {
        super(message, cause);
    }

}
